nav2_smac_planner/src/smac_planner_hybrid.cpp

```diff
@@ -404,6 +414,20 @@ nav_msgs::msg::Path SmacPlannerHybrid::createPlan(
   pose.pose.orientation.z = 0.0;
   pose.pose.orientation.w = 1.0;
 
+  // Corner case of start and goal being on the same cell
+  if (std::floor(mx_start) == std::floor(mx_goal) &&
+    std::floor(my_start) == std::floor(my_goal))
+  {
+    pose.pose = start.pose;
+    // if we have a different start and goal orientation, set the unique path pose to the goal
+    // orientation
+    if (start.pose.orientation != goal.pose.orientation) {
+      pose.pose.orientation = goal.pose.orientation;
```

Comment:
Why not just always do this? I don't think we even need the if branch

Reply:
I took inspiration from the structure in smac_planner_2d, but you're right, there is no added value here.